provider/resource_keycloak_identity_provider_token_exchange_scope_permission.go

```diff
@@ -0,0 +1,292 @@
+			},
+			"authorization_idp_resource_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Resource id representing the identity provider, this automatically created by keycloak",
+			},
+			"authorization_token_exchange_scope_permission_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Permission id representing the Permission with scope 'Token Exchange' and the resource 'authorization_idp_resource_id', this automatically created by keycloak, the policy id will be set on this permission",
+			},
+		},
+	}
+}
+
+func setIdentityProviderTokenExchangeScopePermissionClientPolicy(keycloakClient *keycloak.KeycloakClient, realmId, providerAlias string, clients []string) error {
+	identityProviderPermissions, err := keycloakClient.GetIdentityProviderPermissions(realmId, providerAlias)
+	if err != nil {
+		return err
+	}
+
+	realmManagementClient, err := keycloakClient.GetOpenidClientByClientId(realmId, "realm-management")
+	if err != nil {
+		return err
+	}
+
+	tokenExchangeScopedPermissionId, err := identityProviderPermissions.GetTokenExchangeScopedPermissionId()
+	if err != nil {
+		return err
+	}
+
+	permission, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, tokenExchangeScopedPermissionId)
+	if err != nil {
+		return err
+	}
+
+	if len(permission.Policies) == 0 {
+		policyId, err := createClientPolicy(keycloakClient, realmId, realmManagementClient.Id, providerAlias, clients)
+		if err != nil {
+			return err
+		}
+		permission.Policies = []string{policyId}
+		return keycloakClient.UpdateOpenidClientAuthorizationPermission(permission)
+
+	} else if len(permission.Policies) == 1 {
+		openidClientAuthorizationClientPolicy, err := keycloakClient.GetOpenidClientAuthorizationClientPolicy(realmId, realmManagementClient.Id, permission.Policies[0])
+		if err != nil {
+			return err
+		}
+		openidClientAuthorizationClientPolicy.Clients = clients
+		return keycloakClient.UpdateOpenidClientAuthorizationClientPolicy(openidClientAuthorizationClientPolicy)
+
+	} else {
+		return fmt.Errorf("only supports 1 client policy is supporterd on the identity provider token exchange scope based permission, but %d were found", len(permission.Policies))
+	}
+}
+
+func createClientPolicy(keycloakClient *keycloak.KeycloakClient, realmId, realmManagementClientId, providerAlias string, clients []string) (string, error) {
+	openidClientAuthorizationClientPolicy := &keycloak.OpenidClientAuthorizationClientPolicy{
+		RealmId:          realmId,
+		ResourceServerId: realmManagementClientId,
+		Name:             realmId + "_" + providerAlias + "_client_policy",
+		DecisionStrategy: "UNANIMOUS",
+		Logic:            "POSITIVE",
+		Type:             "client",
+		Clients:          clients,
+	}
+	err := keycloakClient.NewOpenidClientAuthorizationClientPolicy(openidClientAuthorizationClientPolicy)
+	if err != nil {
+		if strings.Contains(err.Error(), "409 Conflict") {
```

Comment:
let's make a helper function to detect 409 errors, similar to the one we use for 404 errors: https://github.com/mrparkers/terraform-provider-keycloak/blob/766d7b6ded815cfe5db986a053af2bdc5b78c344/keycloak/error.go#L17-L21

Reply:
:+1:  I will take look on Monday for this